inbox/mailsync/service.py

```diff
@@ -118,24 +117,24 @@ def run(self):
         while self.keep_running:
             retry_with_logging(self._run_impl, self.log)
 
+        kill_all(self.contact_sync_monitors.values())
+        self.log.info(
+            "stopped contact sync monitors", count=len(self.contact_sync_monitors)
+        )
+        kill_all(self.event_sync_monitors.values())
+        self.log.info(
+            "stopped event sync monitors", count=len(self.event_sync_monitors)
+        )
+
         for email_sync_monitor in self.email_sync_monitors.values():
             if email_sync_monitor.delete_handler:
                 email_sync_monitor.delete_handler.kill()
+            kill_all(email_sync_monitor.folder_monitors, block=False)
```

Comment:
`folder_monitors` are children of `email_sync_monitor` and `email_sync_monitor` kills them in the cleanup but it makes sense to start killing them as soon as possible without blocking as it speeds up the process shutdown process. Again order here is not really meaningful from the correctness point of views.